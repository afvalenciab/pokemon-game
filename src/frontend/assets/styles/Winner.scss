.overlay__winner {
  display: flex;
  width: 100%;
  height: 100%;
  position: fixed;
  z-index: 3;
  top: 0;
  left: 0;
  background: rgba(0, 0, 0, .8);
  overflow-x: hidden;
}

.winner__container {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  width: 400px;
  height: auto;
  background: #fff;
  padding: 30px;
  margin: auto;
  animation-name: winner;
  animation-duration: 1s;

  h1 {
    font-size: 30px;
    letter-spacing: 2px;
    color: rgb(185, 151, 24);
  }

  figure {
    margin: 0;
    img {
      vertical-align: top;
      width: 220px;
      height: 220px;
    }
  }

  p {
    font-size: 25px;
    letter-spacing: 1.5px;
    margin: 0;
  }

  button {
    outline: none;
    background: #3C5BA6;
    letter-spacing: 1px;
    color: #fff;
    font-weight: bold;
    border: 0;
    padding: 10px 30px;
    border-radius: 7px;
    margin-top: 20px;
    cursor: pointer;
    transition: .2s;
    box-shadow: inset 0 -4px rgba(0,0,0, .4);
    will-change: transform;
  }

  button:active {
    transform: scale(.9);
  }
}

@keyframes winner {
  0% {
    opacity: 0;
    transform: scale(.3);
  }
  100% {
    opacity: 1;
    transform: scale(1);
  }
};