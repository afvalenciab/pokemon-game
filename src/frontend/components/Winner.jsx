import React from 'react';
import '../assets/styles/Winner.scss';

const Winner = (props) => {
  const { name, pictureDefault, hanldeClickRestart } = props;

  // const handleClick = (event) => {
  //   event.preventDefault();
  //   hanldeClickRestart();
  // };

  return (
    <div className="overlay__winner">
      <section className="winner__container">
        <h1>GANADOR</h1>
        <figure>
          <img src={pictureDefault} alt=""/>
        </figure>
        <p>{name}</p>
        <button onClick={hanldeClickRestart}>Nuevo Desafio</button>
      </section>
    </div>
  );
};

export default Winner;
