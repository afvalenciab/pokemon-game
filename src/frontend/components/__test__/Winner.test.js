import React from 'react';
import { create } from 'react-test-renderer';
import Winner from '../Winner';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('Winner testing', () => {
  test('Match Snapshot', () => {
    const winner = create(<Winner />);
    expect(winner.toJSON()).toMatchSnapshot();
  });

  test('section has .winner__container class', () => {
    const winner = shallow(<Winner />);
    const winnerElm = winner.find('section');
    expect(winnerElm.hasClass('winner__container')).toBe(true);
  });
});