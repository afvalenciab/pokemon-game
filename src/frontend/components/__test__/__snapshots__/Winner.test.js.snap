// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Winner testing Match Snapshot 1`] = `
<div
  className="overlay__winner"
>
  <section
    className="winner__container"
  >
    <h1>
      GANADOR
    </h1>
    <figure>
      <img
        alt=""
      />
    </figure>
    <p />
    <button>
      Nuevo Desafio
    </button>
  </section>
</div>
`;
